### PR TITLE
system checks for mail and environment variables

### LINKED ISSUE #393: Implement system-checks
Implement and include tests using the [system check framework](https://docs.djangoproject.com/en/3.0/topics/checks/)

- [x] Connect to Sendgrid and send a mail using the [Sandbox Mode](https://sendgrid.com/docs/for-developers/sending-email/sandbox-mode/) and verifying HTTP 200 OK Code
- [x] Verify all environment variables that are necessary to run have been set
- [ ] Possible further tests? Mapbox token?

Failure of these tests should quit the application immediately (by returning a suitable Exception)

### PR DESCRIPTION
**Pull request by feeds**
Closes #393

### CODE REVIEW

**Comment by feeds** [comment]
@maltezacharias Where do we add the `python3 manage.py check` to the dockerfile, so it actually gets executed?

**Comment by Baschdl** [comment]
@feeds Commands in the Dockerfile are only executed when building the container and because Docker caches builds this happens only when there are changes in the Dockerfile or requirements.txt.

You probably want to add it to the ./deploy.sh if you want to execute it with each deployment.

File: backend/apps/checks.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
+from django.apps import apps as camelot_apps
```
Comment (Baschdl) [inline backend/apps/checks.py]: Is that a standard name?

File: backend/apps/checks.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+from django.core.checks import register, Error, Warning
+from django.core.checks import Tags as DjangoTags
+from django.conf import settings
+import http.client
+import json
+
+
+def sendgrid_sandbox_mail_works():
```
Comment (Baschdl) [inline backend/apps/checks.py]: How about `sendgrid_sandbox_check_if_mail_works`?

File: backend/apps/checks.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+
+class Tags(DjangoTags):
+    mail_tag = 'mails'
+    env_tag = 'environment'
+
+
+@register(Tags.env_tag)
+def check_env_variables_set(app_configs=None, **kwargs):
+    """Check that all required environment variables (not belonging to emails) are set."""
```
Comment (Baschdl) [inline backend/apps/checks.py:45]: We should add an explanation to the README stating that one has to add a check if they introduce a new variable

**Review by Baschdl — DISMISSED** [review]
@kevihiiin With this change, we should not use the `.env` file and the separate `run/sendgrid.key` file to set the sendgrid key. What is the better option?

File: backend/apps/checks.py (hunk 1 above)
Comment (feeds) [reply backend/apps/checks.py]: well, its copy and past so probably

File: backend/apps/checks.py (hunk 2 above)
Comment (josauder) [reply backend/apps/checks.py]: dein ernie??

**Review by maltezacharias — APPROVED** [review]
Removed sendgrid api key from file, as per @Baschdl comment.
Checked in local deployment, looks perfect to me!

**Pull request merged**